File: 4057-total-waviness-of-numbers-in-range-i/total-waviness-of-numbers-in-range-i.py
class Solution:
    def totalWaviness(self, num1: int, num2: int) -> int:
        def waviness(num):
            x = str(num)
            n = len(x)
            count = 0

            for i in range(1, n - 1):
                if x[i - 1] < x[i] > x[i + 1] or x[i - 1] > x[i] < x[i + 1]:
                    count += 1

            return count

        count = 0
        for num in range(num1, num2 + 1):
            count += waviness(num)

        return count

File: 4057-total-waviness-of-numbers-in-range-i/total-waviness-of-numbers-in-range-i.cs
public class Solution {
    public int TotalWaviness(int num1, int num2) {
        int Waviness(int num) {
            string s = num.ToString();
            int n = s.Length;
            int count = 0;

            for (int i = 1; i < n - 1; i++) {
                if ((s[i - 1] < s[i] && s[i] > s[i + 1]) || (s[i - 1] > s[i] && s[i] < s[i + 1])) {
                    count++;
                }
            }

            return count;
        }

        int total = 0;

        for (int num = num1; num <= num2; num++) {
            total += Waviness(num);
        }

        return total;
    }
}